PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalInternalClient.java

```diff
@@ -59,6 +62,10 @@ void sendRequest(final Context context, final PayPalRequest payPalRequest,
                         ? SETUP_BILLING_AGREEMENT_ENDPOINT : CREATE_SINGLE_PAYMENT_ENDPOINT;
                 String url = String.format("/v1/%s", endpoint);
                 String appLinkReturn = isBillingAgreement ? appLink : null;
+                
+                final String linkType = (isBillingAgreement &&
+                                        ((PayPalVaultRequest) payPalRequest).getEnablePayPalAppSwitch() &&
+                                        braintreeClient.isPayPalInstalled()) ? "universal" : "deeplink";
```

Comment:
If this is only used internally can we call it "applink" instead of "universal"?

Reply:
We unfortunately decided these names on iOS for the venmo app switch project. We pass the same for that flow as well: https://github.com/braintree/braintree_android/blob/01c5f5237b51c0cf5dfa4fa2fe767000513f0b6b/Venmo/src/main/java/com/braintreepayments/api/venmo/VenmoClient.java#L38. We can consider changing it on android but would need to loop in the analytics team since they currently build their analytics around universal/deeplink as the expected values.

Reply:
Ahhh ok disregard then we can leave it "universal"

Reply:
This PR isn't actually sending the `link_type` analytics value. For that we should stick with `universal` or `deeplink`, like Jax called out.

But the code here is just internal and can be whatever names we want. The `universal` naming doesn't have to come into play until the analytics layer ([here](https://github.com/braintree/braintree_android/blob/01c5f5237b51c0cf5dfa4fa2fe767000513f0b6b/BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsClient.kt#L224-L248))

Reply:
What Sammy mentioned is correct. In my mind, I have two options: one is to convert this to a boolean and have the same logic in `BTPayPalClient` (Where I really need the string with values _"universal"_ or _"deeplink"_), or as a second option, modify this method so that it allows me to pass the `linkType` as a parameter from `BTPayPalClient`.

Any other ideas? What do you all think?

Reply:
I'd vote for creating an `enum` like [Sammy mentioned](https://github.com/braintree/braintree_android/pull/1092/files#r1702196392). With an enum the `if/else` logic can be converted to a Kotlin `when()` clause (similar to `switch` in Swift).